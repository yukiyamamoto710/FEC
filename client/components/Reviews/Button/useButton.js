import { useState } from 'react';

export default function useButton(getMoreReviews) {
  const [isAddReview, setIsAddReview] = useState(false);

  const cancelAddReview = () => {
    setIsAddReview(false);
  };

  const handleClickAddReview = () => {
    setIsAddReview(true);
  };

  const handleClickMoreReview = () => {
    getMoreReviews(2);
  };
  return {
    handleClickMoreReview,
    handleClickAddReview,
    cancelAddReview,
    isAddReview,
  };
}
