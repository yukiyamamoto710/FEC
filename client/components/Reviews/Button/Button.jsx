import React from 'react';
import PropTypes from 'prop-types';
import PopOut from '../PopOut/PopOut';
import useButton from './useButton';

const Button = (props) => {
  const {
    rating,
    getMoreReviews,
    addUserReview,
    isMoreReviews,
  } = props;
  const {
    handleClickMoreReview,
    handleClickAddReview,
    cancelAddReview,
    isAddReview,
  } = useButton(getMoreReviews);
  return (
    <div
      data-testid="addmoreBtn"
      className="add_moreBTNContainer"
    >
      {isAddReview
        ? (
          <PopOut
            addUserReview={addUserReview}
            data={rating}
            cancelAddReview={cancelAddReview}
          />
        )
        : null}
      {isMoreReviews === true
        ? (
          <button
            data-testid="moreBtn"
            className="add_moreBTN"
            type="button"
            onClick={handleClickMoreReview}
          >
            MORE REVIEWS
          </button>
        ) : null }
      <button
        data-testid="AddBtn"
        className="add_moreBTN"
        type="button"
        onClick={handleClickAddReview}
      >
        ADD A REVIEWS
      </button>
    </div>
  );
};

Button.propTypes = {
  rating: PropTypes.shape({}),
  isMoreReviews: PropTypes.bool,
  addUserReview: PropTypes.func,
  getMoreReviews: PropTypes.func,
};

Button.defaultProps = {
  rating: {},
  isMoreReviews: true,
  addUserReview: () => (1),
  getMoreReviews: () => (1),
};

export default Button;
