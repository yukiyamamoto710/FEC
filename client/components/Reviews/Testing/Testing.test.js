/**
 * @jest-environment jsdom
 */
import React from 'react';
import renderer from 'react-test-renderer';
import { render, cleanup } from '@testing-library/react';
import Button from './Testing';

afterEach(cleanup);

test('render without crashing', () => {
  const { getByTestId } = render(<Button name="i am button" />);
  expect(getByTestId('Button').innerHTML).toBe('i am button');
});

it('matches snapShot', () => {
  const tree = renderer.create(<Button name="i am button" />).toJSON();
  expect(tree).toMatchSnapshot();
});
