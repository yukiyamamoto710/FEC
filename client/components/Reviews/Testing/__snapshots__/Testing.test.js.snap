// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapShot 1`] = `
<div
  data-testid="Button"
>
  i am button
</div>
`;
